### PR TITLE
Add quick start docker compose example

### PR DESCRIPTION
**Pull request by manfre**
Modified the Dockerfile to create the folders necessary to use the example with a volume mount or local volume bind.

Provide an optional entrypoint that will create the initial admin user based upon provided environment variables.
* GTS_ADMIN_USERNAME
* GTS_ADMIN_EMAIL
* GTS_ADMIN_PASSWORD

### CODE REVIEW

File: scripts/quick-start.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        echo "Initializing Admin user ${GTS_ADMIN_USERNAME}"
+        /gotosocial/gotosocial admin account create --username ${GTS_ADMIN_USERNAME} --email ${GTS_ADMIN_EMAIL} --password '${GTS_ADMIN_PASSWORD}'
+
+        /gotosocial/gotosocial admin account promote --username ${GTS_ADMIN_USERNAME}
+    else
+        echo "Existing admin user ${GTS_ADMIN_USERNAME} found"
+    fi
+fi
+
+/gotosocial/gotosocial server start
```
Comment (tsmethurst) [inline scripts/quick-start.sh:21]: I think this should use `exec`, since we don't want the script to be the parent

File: scripts/quick-start.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+set -e
+
+if [ "$GTS_ADMIN_USERNAME" != "" ]
+then
```
Comment (tsmethurst) [inline scripts/quick-start.sh:6]: I would add some checks here to ensure that the other necessary environment variables are also set

File: example/docker-compose-quickstart/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+version: "3.8"
```
Comment (tsmethurst) [inline example/docker-compose-quickstart/docker-compose.yml:1]: IMO it overcomplicates things to have a separate docker compose for the quick start. I would rather put this in the existing docker-compose file and just comment out the quickstart script variables, with a note to uncomment them if that functionality is desired

**Review by tsmethurst — COMMENTED** [review]
I'm not 100% sure about this PR. In some ways I think it makes things *more* complicated, but I think it could be streamlined. @NyaaaWhatsUpDoc what do you think?

File: docs/installation_guide/docker.md [hunk 4]
```diff
@@ -78,7 +78,7 @@ image: superseriousbusiness/gotosocial:0.3.1
 
 #### Host
 
-Change the `GTS_HOST` environment variable to the domain you are running GoToSocial on.
+Change the `GTS_HOST` environment variable to the domain you are running GoToSocial on. If you want your ActivityPub identity to be different than the host name, set the `GTS_ACCOUNT_DOMAIN` environment variable. For example, your `GTS_HOST` is `gts.example.org`, but you want your user to be `myuser@example.org`, instead of `myuser@gts.example.org`, you would get `GTS_ACCOUNT_DOMAIN` as `example.org`.
```
[the remark this answers is not in the document]
Comment (manfre) [reply docs/installation_guide/docker.md:81]: A fair concern. Would it be helpful if I provided (independent of this PR) documentation related to using `account-domain`? Or would that potentially cause more support issues by making that functionality more accessible before it's potentially fully ready for use?

File: example/docker-compose-quickstart/docker-compose.yml (hunk 3 above)
Comment (manfre) [reply example/docker-compose-quickstart/docker-compose.yml:1]: I opted for a separate compose file to help identify it as being not part of the current, standard flow of running from a shell with local files/folders. Combining and commenting to provide example of the quickstart without the extra mounted volumes works for me.

**Comment by manfre** [comment]
> I'm not 100% sure about this PR. In some ways I think it makes things _more_ complicated, but I think it could be streamlined.

Thank you for taking the time to review my PR. I don't want to add complication or potentially increase the support burden on the project. I'm closing this PR to reduce cognitive load on y'all. Discussion about if this functionality is wanted and how to achieve it without adding complication can continue on [the related issue](https://github.com/superseriousbusiness/gotosocial/issues/978).

**Pull request closed without merging**